cloudinit/net/ephemeral.py

```diff
@@ -404,57 +407,217 @@ def __init__(
         interface,
         ipv6: bool = False,
         ipv4: bool = True,
+        connectivity_urls_data: Optional[List[Dict[str, Any]]] = None,
     ):
+        """
+        Args:
+            distro: The distro object
+            interface: The interface to bring up
+            ipv6: Whether to bring up an ipv6 network
+            ipv4: Whether to bring up an ipv4 network
+            connectivity_urls_data: List of url data to use for connectivity
+                check before attempting to bring up ephemeral networks. If
+                connectivity can be established to any of the urls, then the
+                ephemeral network setup is skipped.
+        """
         self.interface = interface
         self.ipv4 = ipv4
         self.ipv6 = ipv6
         self.stack = contextlib.ExitStack()
         self.state_msg: str = ""
         self.distro = distro
+        self.connectivity_urls_data = connectivity_urls_data
 
     def __enter__(self):
-        if not (self.ipv4 or self.ipv6):
+        if not self.ipv4 and not self.ipv6:
             # no ephemeral network requested, but this object still needs to
             # function as a context manager
             return self
         exceptions = []
         ephemeral_obtained = False
+
+        # short-circuit if we already have connectivity to IMDS
+        if imds_url := _check_connectivity_to_imds(
+            self.connectivity_urls_data
+        ):
+            LOG.debug(
+                "We already have connectivity to IMDS at %s, skipping DHCP.",
```

Comment:
This debug statement and the one logged when `_check_connectivity_to_imds()` returns true are redundant. One should suffice.

Reply:
Yeah I noticed that as well. Will fix.